specification/operationsmanagement/resource-manager/Microsoft.OperationsManagement/2015-11-01-preview/OperationsManagement.json

```diff
@@ -0,0 +1,422 @@
+    ],
+    "security": [
+        {
+            "azure_auth": [
+                "user_impersonation"
+            ]
+        }
+    ],
+    "securityDefinitions": {
+        "azure_auth": {
+            "type": "oauth2",
+            "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+            "flow": "implicit",
+            "description": "Azure Active Directory OAuth2 Flow",
+            "scopes": {
+                "user_impersonation": "impersonate your user account"
+            }
+        }
+    },
+    "paths": {       
+        "/subscriptions/{subscriptionId}/resourcegroups/{resourceGroupName}/providers/Microsoft.OperationsManagement/solutions/{solutionName}": {
+            "put": {
+                "tags": [
+                    "Solution"
+                ],
+                "x-ms-examples": {
+                     "SolutionCreate": { "$ref": "./examples/SolutionCreate.json" }
+                },
+                "operationId": "OperationsManagement_CreateSolution",
+                "summary": "Creates Solution.",
+                "description": "Creates or updates the Solution.",
+                "parameters": [
+                    {
+                        "$ref": "#/parameters/SubscriptionIdParameter"
+                    },
+                    {
+                        "$ref": "#/parameters/ResourceGroupNameParameter"
+                    },                    
+                    {
+                        "$ref": "#/parameters/ApiVersionParameter"
+                    },
+                    {
+                        "$ref": "#/parameters/SolutionNameParameter"
+                    },
+                    {
+                        "name": "parameters",
+                        "in": "body",
+                        "required": true,
+                        "schema": {
+                            "$ref": "#/definitions/Solution"
+                        },
+                        "description": "The parameters required to create OMS Solution."
+                    }
+                ],
+                "responses": {
+                    "200": {
+                        "description": "OK response definition.",
+                        "schema": {
+                            "$ref": "#/definitions/Solution"
+                        }
+                    },
+                    "default": {
+                        "schema": {
+                            "$ref": "#/definitions/CodeMessageError"
+                        },
+                        "description": "Error response definition."
+                    }
+                }
+            },
+            "delete": {
+                "tags": [
+                    "Solution"
+                ],
+                 "x-ms-examples": {
+                     "SolutionDelete": { "$ref": "./examples/SolutionDelete.json" }
+                },
+                "operationId": "OperationsManagement_DeleteSolution",
```

Comment:
Should be Solution_Delete.

Reply:
If I make it singular it gives me warning since I have Solution in definition section. Should I use format like Solutions_Get, Solutions_Delete, etc.?

Reply:
You are correct the noun here should be plural, see the [guidelines](https://github.com/Azure/azure-rest-api-specs/blob/current/documentation/openapi-authoring-automated-guidelines.md#r2063-operationidnounconflictingmodelnames).  Good catch and sorry for the confusion.

Reply:
No Worries. Thanks.